Add Product Recipes Template (#2741)
mattermost/docs: source/administration/product-receipe-template.rst

> Product Recipe template

Reviewer: ```suggestion
Product Recipe Template
```

========================

> This template defined how a product recipe can be added for the depending

Reviewer: ```suggestion
This template defines how a product recipe can be added for the depending
```

people.

Problem Statement
-------------------

Define the problem statement. The problem statement could be anything from adding
an architecture setup or installations.

> The problem statement should be written from user perspective and from organisation

Reviewer: ```suggestion
The problem statement should be written from a user perspective and from an organization
```

perspective.

> The user perspective might include

Reviewer: ```suggestion
The user perspective might include:
```

- The requirements of the user
- The environment of the user

> - The dependecies the user is having

Reviewer: ```suggestion
- The dependencies the user is having
```

> - The user case user would run the installation in

Reviewer: ```suggestion
- The use case the user would run the installation in
```

> The organisation perspective might include

Reviewer: ```suggestion
The organisation perspective might include:
```

> - How much effort the organisations adds in implementing the solution

Reviewer: ```suggestion
- How much effort the organization adds in implementing the solution
```

- How many work hours are put on implementing the solution for the user

> - Listing team putting the solution for the user

Reviewer: Not sure what this means

Author: Teams involved in enabling the solution for the users? Let's say if an installation is happening, how many levels of escalation can be present.

> - Dependecies with ongoing installations

Reviewer: ```suggestion
- Dependencies with ongoing installations
```

> Pre-installation/pre-configuration notes

Reviewer: ```suggestion
Pre-Installation/Pre-Configuration Notes
```

----------------------------------------

Define the pre-installation information. Projects installation has pre-installation

> steps describing the things you should setup when installating a requirement.

Reviewer: "Projects installation has pre-installation
steps describing the things you should setup when installating a requirement."

Would you be open to clarifying this sentence?

> Breifly describe the pre-installation or pre-configuration notes. Things you

Reviewer: ```suggestion
Briefly describe the pre-installation or pre-configuration notes. Things you
```

> might include

Reviewer: ```suggestion
might include:
```

- Environment information
- Version information
- Release information

> - Dependecy information

Reviewer: ```suggestion
- Dependency information
```

- Rollback information
- Line of Business information

> Instructions Setup and Configuring the Solution

Reviewer: Would you be open to clarifying the heading?

Author: The actual instructions and how the solution can be implemented.

Author: There is a ticker for this, I picked these works from the ticket.

-----------------------------------------------

> Define the instructions setup instructions. The instructions should be detailed

Reviewer: Please clarify the opening sentence.

and have everything about setting up and configuring the solution. Things you

> might include

Reviewer: ```suggestion
might include:
```

- Installation instructions

> - Trouble shooting instructions

Reviewer: ```suggestion
- Troubleshooting instructions
```

- Testing the installation
- Testing the installation in multiple environments

> Defining Recipe Category For Search and Retrival

Reviewer: ```suggestion
Defining Recipe Category For Search and Retrieval
```
?

------------------------------------------------

Define the categories and tag for the users. The users should be easily search
and retrieve the receipe. Add tech, line of business, architecture, environement

> information.

Reviewer: Please help make this paragraph a little clearer.

Author: If you can tell me what tags you have in your organisation for this sort of implementation, I can include the details. I added few generic things in these lines that could let any member of mattermost org implement a product recipe and add proper tags in the documentation.
